lib/ansible/plugins/action/template.py

```diff
@@ -140,6 +146,9 @@ def run(self, tmp=None, task_vars=None):
             old_vars = self._templar._available_variables
             self._templar.set_available_variables(temp_vars)
             resultant = self._templar.do_template(template_data, preserve_trailing_newlines=True, escape_backslashes=False)
+            # if encode is set, we change template result encoding
+            if encode is not None:
+                resultant = resultant.encode(encode)
```

Comment:
We do not want to do it this way because it means resultant could be a text string or could be a byte string later on.  As the internal APIs evolve one of those two cases is sure to throw an error.  I think that we probably need to leave the output as text here and convert to bytes later.